adoc/chapters/programming_interface.adoc

```diff
@@ -288,161 +288,213 @@ context doesn't match the target backend.
-in <<table.specialmembers.common.reference>> and
-<<table.hiddenfriends.common.reference>> respectively.
+The following <<sycl-runtime>> classes provide <<reference-semantics>>:
+
+* [api]#accessor#
+* [api]#buffer#
+* [api]#context#
+* [api]#device#
+* [api]#device_image#
+* [api]#event#
+* [api]#host_accessor#
+* [api]#host_sampled_image_accessor#
+* [api]#host_unsampled_image_accessor#
+* [api]#kernel#
+* [api]#kernel_id#
+* [api]#kernel_bundle#
+* [api]#local_accessor#
+* [api]#platform#
+* [api]#queue#
+* [api]#sampled_image#
+* [api]#sampled_image_accessor#
+* [api]#stream#
+* [api]#unsampled_image#
+* [api]#unsampled_image_accessor#
+
+[[sec:reference-semantics-reqs]]
+==== Interface requirements
+
+Each <<sycl-runtime>> class that provides <<reference-semantics>> must conform
+to the following requirements, where [code]#T# is the runtime class type:
+
+* _Copyability:_ [code]#T# must be copy constructible and copy assignable in the
+  host application.
+  If [code]#T# is a valid kernel argument, [code]#T# must also be copy
+  constructible and copy assignable within SYCL kernel functions.
+  Any instance of [code]#T# that is a copy of another instance must behave as-if
+  it were the original instance and must represent the same underlying
+  <<native-backend-object>>, where applicable.
+  Any action performed on such a copy must behave as-if it were also performed
+  on the original instance.
+* _Destructibility:_ [code]#T# must be destructible in the host application.
+  If [code]#T# is a valid kernel argument, [code]#T# must also be destructible
+  within SYCL kernel functions.
+  When any instance of [code]#T# is destroyed, any behavior specific to
+  [code]#T# that is specified as performed on destruction is only performed if
+  this instance is the last remaining host copy, in accordance with the above
+  definition of a copy.
+* _Movability:_ [code]#T# must be move constructible and move assignable in the
+  host application.
+  If [code]#T# is a valid kernel argument, [code]#T# must also be move
+  constructible and move assignable within SYCL kernel functions.
+  Any instance of [code]#T# that is the result of a move from another instance
+  must behave as-if it were the original instance and must represent the same
+  underlying <<native-backend-object>>, where applicable.
+  Move construction and move assignment render the moved-from instance of
+  [code]#T# invalid.
+* _Equality:_ [code]#T# must be equality comparable in the host application.
+  Equality between two instances of [code]#T# (i.e., [code]#a == b#) must be
+  true if one instance is a copy of the other, and non-equality between two
+  instances of [code]#T# (i.e., [code]#a != b#) must be true if neither instance
+  is a copy of the other, in accordance with the above definition of a copy.
+  An instance of [code]#T# that has been invalidated by a move operation is not
+  considered to be a copy of any other [code]#T# instance.
```

Comment:
The text originally said,

> unless either instance has become invalidated by a move operation

which I couldn't parse in context. I *think* the sentence I've added here captures the intent.